src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +309,127 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
+                    value = statusFileContents.Slice(0, nonUnitSliceLength);
+                    valueParsed = int.TryParse(value, out results.Pid);
+                }
+#endif
+                if (title.SequenceEqual("VmHWM".AsSpan()))
+                {
+                    value = statusFileContents.Slice(0, unitSliceLength);
+                    valueParsed = ulong.TryParse(value, out results.VmHWM);
+                }
+                else if (title.SequenceEqual("VmRSS".AsSpan()))
+                {
+                    value = statusFileContents.Slice(0, unitSliceLength);
+                    valueParsed = ulong.TryParse(value, out results.VmRSS);
+                }
+                else if (title.SequenceEqual("VmData".AsSpan()))
+                {
+                    value = statusFileContents.Slice(0, unitSliceLength);
+                    valueParsed = ulong.TryParse(value, out ulong vmData);
+                    results.VmData += vmData;
+                }
+                else if (title.SequenceEqual("VmSwap".AsSpan()))
+                {
+                    value = statusFileContents.Slice(0, unitSliceLength);
+                    valueParsed = ulong.TryParse(value, out results.VmSwap);
+                }
+                else if (title.SequenceEqual("VmSize".AsSpan()))
+                {
+                    value = statusFileContents.Slice(0, unitSliceLength);
+                    valueParsed = ulong.TryParse(value, out results.VmSize);
+                }
+                else if (title.SequenceEqual("VmPeak".AsSpan()))
+                {
+                    value = statusFileContents.Slice(0, unitSliceLength);
+                    valueParsed = ulong.TryParse(value, out results.VmPeak);
+                }
+                else if (title.SequenceEqual("VmStk".AsSpan()))
+                {
+                    value = statusFileContents.Slice(0, unitSliceLength);
+                    valueParsed = ulong.TryParse(value, out ulong vmStack);
+                    results.VmData += vmStack;
+                }
+
+                Debug.Assert(valueParsed);
+                statusFileContents = statusFileContents.Slice(endIndex + 1);
+            }
+
+            results.VmData *= 1024;
+            results.VmPeak *= 1024;
+            results.VmSize *= 1024;
+            results.VmSwap *= 1024;
+            results.VmRSS *= 1024;
+            results.VmHWM *= 1024;
+            result = results;
+            return true;
+        }
+
+        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, out string fileContents)
+        {
+            try
+            {
+                using (var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
+                {
+                    fileContents = reusableReader.ReadAllText(fileStream);
+                    return true;
+                }
+            }
+            catch (IOException)
```

Comment:
when reading `/proc/<pid>/maps` we also catch `UnauthorizedAccessException` . should we do that here?